force-app/main/default/classes/NotificationsTrayController.cls

```diff
@@ -257,7 +257,12 @@ public with sharing class NotificationsTrayController {
 
         domainPath = (domainPath.endsWith('/s')) ? domainPath.left(domainPath.length() - 2) : domainPath;
         
-        domainPath = domainPath.replace(domainPathPrefix, '');
+        //domainPath = domainPath.replace(domainPathPrefix, '');
+        
+        if (domainPath.endsWith(domainPathPrefix)) {
```

Comment:
Hi @gtabboud so in my org what it was doing is my org url starts like this https://loccrm--   and endswith /loc. so end result what https://crm--    so basically it was replacing both the /loc.   but in second commit I have replaced String domainPath = System.Site.getBaseSecureUrl();
String domainPathPrefix = System.Site.getPathPrefix();    with String domainPath = Url.getOrgDomainUrl().toExternalForm();  because it was through error chatter connect api not enabled if we use org url instead of site url it works fine

Reply:
Ok so I took a slightly different approach which I will push to the repo on the next commit:

```
domainPath = domainPath.substringBeforeLast('/s');
        
domainPath = domainPath.substringBeforeLast(domainPathPrefix);
```

Reply:
this won't be needed if you use `Url.getOrgDomainUrl().toExternalForm()` if we use this  `System.Site.getBaseSecureUrl(); ` it gives an error `The Chatter Connect API is not enabled for this organization or user type.","errorCode":"API_DISABLED_FOR_ORG`

Reply:
Going off of memory here, but I remember trying that, and the problem I ran into with that is when there was a custom domain configured for the site. The user's session isn't valid for the domain returned by Url.getOrgDomainUrl().toExternalForm().

Reply:
Hi @gtabboud thanks much and so is there any permission we have to give to resolve this error `Chatter Connect API is not enabled for this organization or user type.`

Reply:
Url.getOrgDomainUrl().toExternalForm() <-- actually this will probably never work for external users. Per the documentation, it says it returned the "my domain" of the org, which is the internal domain...external users will not have valid sessions for that domain.

As for that chatter API error, Is chatter not enabled in your org?

Reply:
Its enabled and I have searched for other permission related to it and  didn't anything also  same problem was occurring for Sebastian as well

Reply:
I thought his problem was fixed with your workaround?

Reply:
yes by using this Url.getOrgDomainUrl().toExternalForm()

Reply:
I thought it was with just fixing the replace I was doing... did Url.getOrgDomainUrl().toExternalForm() work for external users for you or did you just test with an admin user?

Reply:
Yes for external users I tested we are using external users

Reply:
So it looks like the "API Enabled" permission is required on the profile for the external user.

Reply:
For me its not working using external site domain even though I have provided Api enabled through permission set

Reply:
what's the error that you get?

Reply:
In debug logs  `The Chatter Connect API is not enabled for this organization or user type.","errorCode":"API_DISABLED_FOR_ORG`

Reply:
I'm not sure why this is happening for you then. I was able to reproduce that error and when I turn that permission on, it goes away.

Can you post your site's url, and what the callout url is from your debug logs when you get that error?

Reply:
Do you by any chance have a site at the root domain https://loccrm--locdev.sandbox.my.site.com?

Reply:
No

Reply:
is your site actually active or is it still in preview?

Reply:
its active

Reply:
I'm stumped, this works for me in 3 different orgs all via the site url.

Reply:
For now In my code will use org domain if I find anything else will update on this  and thanks so much for the help

Reply:
yes

Reply:
<img width="319" alt="Screenshot 2024-03-05 at 10 55 13 PM" src="https://github.com/salesforce-experiencecloud/notificationbell/assets/84844473/309ae0c7-043c-4e81-9681-e5ef35d42423">